scene/3d/physics_body_3d.cpp

```diff
@@ -502,6 +502,18 @@ void RigidBody3D::_body_state_changed(PhysicsDirectBodyState3D *p_state) {
 
 	GDVIRTUAL_CALL(_integrate_forces, p_state);
 
+	set_global_transform(p_state->get_transform());
+
+	linear_velocity = p_state->get_linear_velocity();
+	angular_velocity = p_state->get_angular_velocity();
+
+	inverse_inertia_tensor = p_state->get_inverse_inertia_tensor();
+
+	if (sleeping != p_state->is_sleeping()) {
+		sleeping = p_state->is_sleeping();
+		emit_signal(SceneStringNames::get_singleton()->sleeping_state_changed);
+	}
+
```

Comment:
It might be a good idea to move all this to a new private method instead, like a `void _sync_body_state(PhysicsDirectBodyState3D *p_state)`, so that you can't accidentally miss one or the other when adding to it. Same goes for the other classes as well.